### PR TITLE
New installation guide

### PR DESCRIPTION
**Pull request by honzajavorek**
#### :rocket: Why this change?

The guide is incomplete and doesn't capture many pitfalls users can meet. We're now also adding an official Docker support. This PR also adds a support for tabbed content to the docs as a side-effect.

#### :eyes: Preview

screen shot 2018-11-27 at 15 28 53

#### :memo: Related issues and Pull Requests

- Resolve the core of https://github.com/apiaryio/dredd/issues/904
- Close https://github.com/apiaryio/dredd/issues/999
- Close https://github.com/apiaryio/docker-base-images/issues/130
- Mitigate https://github.com/apiaryio/dredd/issues/659

#### 📈  Next steps

- Write a dedicated page about Continuous Integration and tackle https://github.com/apiaryio/dredd/issues/353
- Revamp quickstart as it is tightly connected to the first time user experience - https://github.com/apiaryio/dredd/issues/904 (this resolves the rest of https://github.com/apiaryio/dredd/issues/904)
- Document using Docker with the server process and the hook handlers - https://github.com/apiaryio/dredd/issues/755 and https://github.com/apiaryio/docker-base-images/issues/149

#### :white_check_mark: What didn't I forget?

- [x] To write docs
- [ ] To write tests
- [x] To put [Conventional Changelog](https://dredd.org/en/latest/internals.html#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -4,105 +4,226 @@
...
 
-1. `Download Node.js <https://nodejs.org/en/download/>`__ from the official website and install Node.js using the downloaded installer.
-2. Make sure both ``node --version`` and ``npm --version`` work in your Command Prompt.
+      - `Download Node.js <https://nodejs.org/en/download/>`__ from the official website and install it using the downloaded installer
+      - Make sure both ``node --version`` and ``npm --version`` work in your Command Prompt
+      - Node.js needs to be at least version 6
 
-Linux
-~~~~~
+.. note::
```
Comment (JackuB) [inline docs/installation.rst]: I think from a doc standpoint - you should either recommend it or not mention it

File: docs/installation.rst [hunk 2]
```diff
@@ -4,105 +4,226 @@
...
+   You can also use `nvm <https://github.com/creationix/nvm>`__ to install Node.js on your computer.
 
-1. `Install Node.js as system package <https://nodejs.org/en/download/package-manager/>`__.
-2. Make sure both ``node --version`` and ``npm --version`` work in your Terminal.
+.. warning::
+   If your internet connection is restricted (VPN, firewall, proxy), you need to run following additional commands to configure npm:
 
-Pro Tips
-~~~~~~~~
+   .. code-block:: text
```
Comment (JackuB) [inline docs/installation.rst]: Might be better to link to the npm's docs on this use case as the npm commands could change https://docs.npmjs.com/misc/config#https-proxy

File: docs/installation.rst [hunk 3]
```diff
@@ -4,105 +4,226 @@
...
--  **For development**, always go with the latest version.
--  **For testing in CI** (:ref:`what’s CI? <continuous-integration>`), always pin your Dredd version to a specific number and upgrade to newer releases manually (but often!).
+Git is needed to install some of Dredd's dependencies (:ghissue:`gavel.js#83`).
+
+.. tabs::
+
+   .. group-tab:: macOS
+
+      - First try whether Git isn't already available by running ``git --version``
+      - If not and you're using `Homebrew <https://brew.sh/>`__, run ``brew install git``
```
Comment (JackuB) [inline docs/installation.rst]: According to this https://git-scm.com/book/en/v2/Getting-Started-Installing-Git it should be available by default.

File: docs/installation.rst [hunk 4]
```diff
@@ -4,105 +4,226 @@
...
 
 .. code-block:: text
 
-   npmERR! Cannot read property 'path' of null
-   npmERR!code ECONNRESET
-   npmERR!network socket hang up
+   npm install dredd --global --no-optional
+
+.. warning::
+   If you get ``EACCES`` permissions errors, run the command again with ``sudo`` or try `one of the officially recommended solutions <https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally>`__.
```
Comment (JackuB) [inline docs/installation.rst]: Would advise to show the correct way of resolving it first, then sudo

File: docs/installation.rst (hunk 2 above)
Comment (honzajavorek) [reply docs/installation.rst]: I decided to keep the commands there. The official docs do not contain easy-to-copy-paste examples and people who don't know how to work with npm might get just more confused on what they're supposed to do to fix the problem. I'll link the docs tho.

Comment (honzajavorek) [reply docs/installation.rst]: (These proxy sections are added based on real people struggling with it and reporting me back that the installation guide doesn't work for them... and they were not very skilled with npm.)

**Comment by honzajavorek** [comment]
@JackuB @kylef comments addressed

**Comment by honzajavorek** [comment]
Thanks! 📝

**Pull request merged**